modules/platform/nuxeo-platform-types/src/main/java/org/nuxeo/ecm/platform/types/Type.java

```diff
@@ -106,85 +85,63 @@ public void addSubType(List<SubType> subTypes) {
     @XNode("category")
     protected String category;
 
+    @XNodeMap(value = "views/view", key = "@id", type = HashMap.class, componentType = TypeView.class)
     protected Map<String, TypeView> views;
 
     @XNodeList(value = "actions/action", type = String[].class, componentType = String.class)
+    @XMerge(defaultAssignment = false) // compat
     protected String[] actions;
 
     @XNodeMap(value = "layouts", key = "@mode", type = HashMap.class, componentType = Layouts.class)
-    Map<String, Layouts> layouts;
+    @XMerge(value = "layouts@append", defaultAssignment = false) // compat
+    protected Map<String, Layouts> layouts;
 
     @XNodeMap(value = "contentViews", key = "@category", type = HashMap.class, componentType = DocumentContentViews.class)
+    @XMerge(value = "contentViews@append", defaultAssignment = false) // compat
     protected Map<String, DocumentContentViews> contentViews;
 
-    // for bundle update::
-    @XNode("@remove")
-    protected boolean remove = false;
+    // needed by xmap
+    public Type() {
+    }
 
-    public String[] getActions() {
-        return actions;
+    // helper for tests
+    public Type(String id, String label) {
+        this.id = id;
+        this.label = label;
     }
```

Comment:
Not sure it is used.

Reply:
It's useful only for tests on JSF side: https://github.com/nuxeo/nuxeo-jsf-ui/pull/100/commits/ac6e9322109f54a4679c25a6192e499574ac5baa